pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -198,6 +201,10 @@ public ProducerImpl(PulsarClientImpl client, String topic, ProducerConfiguration
         this.userProvidedProducerName = StringUtils.isNotBlank(producerName);
         this.partitionIndex = partitionIndex;
         this.pendingMessages = createPendingMessagesQueue();
+        // Replication needs to be paused when a message can not be replicated due to failed schema registration.
+        // Otherwise, it may cause an out-of-order issue, and it may lead to a messages lost issue if users enabled
+        // deduplication on the remote side.
+        this.pauseSendingToPreservePublishOrderOnSchemaRegFailure = conf.isReplProducer();
```

Comment:
Main change 3:

Replication should not skip the messages who contain a broken schema